network/src/protocols/mod.rs

```diff
@@ -148,6 +148,8 @@ impl CKBProtocol {
             protocol_name: format!("/ckb/{}", protocol_name),
             supported_versions: {
                 let mut versions: Vec<_> = versions.to_vec();
+                // TODO: https://github.com/rust-lang/rust-clippy/issues/6001
+                #[allow(clippy::unnecessary_sort_by)]
```

Comment:
I think this lint can by removed by:
```
versions.sort_by_key(|v| std::cmp::Reverse(v));
```

Reply:
I have tried this way of writing, but it needs clone

Reply:
aha, I didn't realize that `sort_by_key` returns an owned instead of ref, after re-read original code, I think we may use sort and reverse, which avoids cloning twice in `versions.to_vec()`:
```
                let mut versions: Vec<_> = versions.to_vec();
                versions.sort();
                versions.reverse();
                versions
```

Reply:
This can avoid `allow(clippy::unnecessary_sort_by)`, but actually do one more traversal. Although performance is not the most critical during startup, we should not change the correct code because of the wrong prompt